lib/pipelineFactory.js

```diff
@@ -42,8 +42,22 @@ class PipelineFactory extends BaseFactory {
         const modelConfig = hoek.applyToDefaults({
             createTime: (new Date()).toISOString()
         }, config);
+        const admin = config.admins[0];
 
-        return super.create(modelConfig);
+        return Pipeline.getScmRepo(admin, this.scmPlugin, config.scmUrl)
```

Comment:
I don't really understand why this is something the pipeline model has to be a part of, other than it is a convenient place to have static methods to get a user model. 

On the other hand, the pipelineFactory could have a method that fetches a user model, and scmPlugin could have a method `getRepoData(userModel, scmUrl)` or whatever that does the lookup.